comment/models/comments.py

```diff
@@ -49,10 +53,29 @@ def _set_unique_urlhash(self):
             while self.__class__.objects.filter(urlhash=self.urlhash).exists():
                 self.urlhash = self.__class__.objects.generate_urlhash()
 
+    def _set_email(self):
+        if self.user:
+            self.email = self.user.email
+
     def save(self, *args, **kwargs):
         self._set_unique_urlhash()
+        self._set_email()
         super(Comment, self).save(*args, **kwargs)
 
+    def get_url(self, request):
+        page_url = self.content_object.get_absolute_url()
+        if not page_url.endswith('/'):
+            page_url += '/'
+        comments_per_page = settings.COMMENT_PER_PAGE
+        if comments_per_page:
+            qs_all_parents = self.__class__.objects.filter_parents_by_object(
+                self.content_object, include_flagged=is_comment_moderator(request.user)
+                )
+            position = qs_all_parents.filter(posted__gte=self.posted).count() + 1
+            if position > comments_per_page:
+                page_url += '?page=' + str((position // comments_per_page) + 1)
+        return page_url + '#' + self.urlhash
```

Comment:
although this works well, it is currently not used if the comment is hidden as a reply. the url is correct but since the element is hidden, it doesn't show.

Reply:
Nice work! This is not a big deal. Since this url leads to the parten comment, the user will be aware that their comment in the replies section.

Reply:
no it doesn't lead anywhere because the id is currently not present(hidden).

Reply:
this also assumes the comments are sorted date wise although we may or the user may choose to sort them based on other parameter for e.g. replies or likes etc

Reply:
Then we should make the url always leads to the parent comment. This will make it easier.

Reply:
this can be a solution for the time being but all of this conjecture is based when comments are sorted in a definite order. I see currently we may add this option a template tag to include order if the developer want.

Reply:
Yeah this can be implemented in a separate ticket. I would avoid passing variables to template tags onwards since the settings is available to us without hitting the root settings on every use.

Reply:
in an ideal world i would like to highlight the container just as github does while moving to a discuss node when an appropriate url is entered.

Reply:
okay, so i have basically copied their CSS for highlighting the target element. since, i am almost unskilled to judge the correct colour you may pick and choose one if you wish to change. i will commit the changes when they are ready